llama-index-integrations/storage/chat_store/llama-index-storage-chat-store-dynamodb/llama_index/storage/chat_store/dynamodb/base.py

```diff
@@ -155,13 +159,25 @@ def __init__(
                 else botocore_config
             )
             session = boto3.Session(**session_kwargs)
+            async_session = aioboto3.Session(**session_kwargs)
         except ImportError:
             raise ImportError(
                 "boto3 package not found, install with 'pip install boto3"
             )
 
         self._client = session.resource("dynamodb", config=config, **resource_kwargs)
         self._table = self._client.Table(table_name)
+        asyncio.run(
+            self.init_async_table(table_name, async_session, config, **resource_kwargs)
```

Comment:
Its a little annoying to put asyncio calls in the `__init__` -- is there a way we can lazily call this function, and skip the `asyncio.run()` ?

Reply:
moved the `init_async_table` to be called separately, with checks when calling the async versions.